provider-assignments/src/java/org/sakaiproject/archiver/provider/AssignmentArchiver.java

```diff
@@ -0,0 +1,455 @@
+		@Setter
+		String feedbackComment;
+
+		@Setter 
+		String feedbackText;
+
+		@Setter
+		String gradedBy;
+		
+		public SimpleFeedback(AssignmentSubmission submission) {
+			this.feedbackComment = submission.getFeedbackComment();
+			this.feedbackText = submission.getFeedbackText();
+			this.gradedBy = submission.getGradedBy();
+		}
+	}
+	
+	/**
+	 * Simplified helper class to represent an individual submission for an assignment
+	 */
+	private class SimpleSubmission {
+
+		@Setter 
+		String status;
+
+		@Setter
+		String submittedText;
+
+		@Setter
+		String submitterId;
+
+		@Setter
+		List<String> groupSubmitterIds;
+
+		@Setter
+		String timeSubmitted;
+
+		public SimpleSubmission(AssignmentSubmission submission, boolean group) {
+			this.status = submission.getStatus();
+			this.submittedText = submission.getSubmittedText();
+			this.submitterId = submission.getSubmitterId();
+			this.timeSubmitted = submission.getTimeSubmittedString();
+			if (group) {
+				this.groupSubmitterIds = submission.getSubmitterIds();
+			}
+		}
+	}
+
+	/**
+	 * Simplified helper class to represent an individual assignment item in a site
+	 */
+	private class SimpleAssignment {
+
+		@Setter
+		private String id;
+
+		@Setter
+		private String timeOpen;
+
+		@Setter
+		private String timeDue;
+
+		@Setter
+		private String lateAfter;
+
+		@Setter
+		private String acceptUntil;
+
+		@Setter
+		private boolean draft;
+
+		@Setter
+		private String creator;
+
+		@Setter
+		private List authors;
+
+		@Setter
+		private String instructions;
+
+		@Setter
+		private String authorLastModified;
+
+		@Setter
+		private String title;
+
+		@Setter
+		private String status;
+
+		@Setter
+		private Collection groups;
+
+		@Setter
+		private String access;
+
+		@Setter
+		private String gradeScale;
+
+		@Setter
+		private String gradeScaleMaxPoints;
+
+		@Setter
+		private String submissionType;
+
+		@Setter
+		private String modelAnswerText;
+
+		@Setter
+		private String privateNoteText;
+
+		@Setter
+		private String allPurposeItemText;
+
+		@Setter
+		private SimpleGradebookItem gradebookItemDetails;
+
+		public SimpleAssignment(Assignment a) {
+			if (a == null){
+				return;
+			}
+
+			// These fields can simply be copied over
+			this.id = a.getId();
+			this.timeOpen = a.getOpenTimeString();
+			this.timeDue = a.getDueTimeString();
+			this.lateAfter = a.getDropDeadTimeString();
+			this.acceptUntil = a.getCloseTimeString();
+			this.draft = a.getDraft();
+			this.creator = a.getCreator();
+			this.authors = a.getAuthors();
+			this.authorLastModified = a.getAuthorLastModified();
+			this.title = a.getTitle();
+			this.status = a.getStatus();
+			this.access = a.getAccess().toString();
+			
+			// See if there are groups submissions for this assignment
+			if (a.isGroup()) {
+				this.groups = a.getGroups();
+			}
+
+			// See if there is a gradebook item associated with this assignment
+			SimpleGradebookItem gradebookItem = getGradebookFields(a);
```

Comment:
whats the purpose of this? just to get the info about any linked gradebook item?